functests/sctp/sctp.go

```diff
@@ -483,3 +488,59 @@ func setupEgress(namespace, fromPod, toPod string, port int32) error {
 	_, err := client.Client.NetworkPolicies(namespace).Create(policy)
 	return err
 }
+
+func findSCTPNodeSelector() {
+	mcList, err := client.Client.MachineConfigs().List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mc := range mcList.Items {
+		if doesThisEnablesSctp(mc) == true {
+			findMCPByLabel(mc.ObjectMeta.Labels["machineconfiguration.openshift.io/role"])
+			break
+		}
+	}
+}
+
+func doesThisEnablesSctp(mc mcfgv1.MachineConfig) bool {
+	firstPathFound := false
+	secondPathFound := false
+
+	for _, file := range mc.Spec.Config.Storage.Files {
+		if file.Path == sctpBlacklistPath {
+			firstPathFound = true
+		} else if file.Path == sctpLoadPath {
+			secondPathFound = true
+		}
+	}
+
+	return firstPathFound && secondPathFound
+}
+
+func findMCPByLabel(mcLabel string) {
+	mcpList, err := client.Client.MachineConfigPools().List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mcp := range mcpList.Items {
+		for _, lsr := range mcp.Spec.MachineConfigSelector.MatchExpressions {
+			for _, value := range lsr.Values {
+				if value == mcLabel {
+					for key, label := range mcp.Spec.NodeSelector.MatchLabels {
+						sctpNodeSelector = key + "=" + label
```

Comment:
Can you just return the mcp here (and an error if not found) and use it outside?
In this way, this function would not have side effects. I'd like to see the nodeList set with the result of `findSctpNodeSelector`